<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>855</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>680</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color:#009999;</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>10</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">font-size:18px;
font-weight:bold;</string>
    </property>
    <property name="text">
     <string>MY FIREWALL</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>190</y>
      <width>741</width>
      <height>181</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border:3px solid white;
font-size:15px;
background-color:#336699;
</string>
    </property>
    <property name="title">
     <string>CREATE RULE</string>
    </property>
    <widget class="QComboBox" name="comboBoxChain">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>50</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none;
background-color:white;
color:black;</string>
     </property>
     <property name="frame">
      <bool>false</bool>
     </property>
     <item>
      <property name="text">
       <string>--</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>INPUT</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>OUTPUT</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>FORWARD</string>
      </property>
     </item>
    </widget>
    <widget class="QComboBox" name="comboBoxOption">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>120</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none;
background-color:white;
padding-left:10px;
color:black;</string>
     </property>
     <item>
      <property name="text">
       <string>--</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>-A</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>-I</string>
      </property>
     </item>
    </widget>
    <widget class="QComboBox" name="comboBoxProtocol">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>50</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none;
background-color:white;
padding-left:10px;
color:black;</string>
     </property>
     <item>
      <property name="text">
       <string>--</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>tcp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>udp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>icmp</string>
      </property>
     </item>
    </widget>
    <widget class="QComboBox" name="comboBoxAction">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>120</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:white;
border:none;
padding-left:10px;
color:black;</string>
     </property>
     <item>
      <property name="text">
       <string>--</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>ACCEPT</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>REJECT</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>DROP</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>30</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none;
font-size:20px;
</string>
     </property>
     <property name="text">
      <string>chain</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>100</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none;
font-size:20px;
</string>
     </property>
     <property name="text">
      <string>option</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>20</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none;
font-size:20px;
</string>
     </property>
     <property name="text">
      <string>protocol</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>100</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none;
font-size:20px;
</string>
     </property>
     <property name="text">
      <string>action</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_create">
     <property name="geometry">
      <rect>
       <x>610</x>
       <y>120</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #009933;
border:none;
border-radius:10px;
color:white;</string>
     </property>
     <property name="text">
      <string>CREATE</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>90</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none;
font-size:20px;
</string>
     </property>
     <property name="text">
      <string>dest port</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>20</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none;
font-size:20px;
</string>
     </property>
     <property name="text">
      <string>source port</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>480</x>
       <y>90</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none;
font-size:20px;
</string>
     </property>
     <property name="text">
      <string>dest IP</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="txt_destIp">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>120</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:white;
border:none;
color:black;</string>
     </property>
     <property name="text">
      <string>--</string>
     </property>
     <property name="maxLength">
      <number>15</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_sPort">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>50</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:white;
border:none;
padding-left:5px;
color:black;</string>
     </property>
     <property name="maximum">
      <number>99999</number>
     </property>
    </widget>
    <widget class="QLineEdit" name="txt_sourceIp">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>50</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:white;
border:none;
color:black;</string>
     </property>
     <property name="text">
      <string>--</string>
     </property>
     <property name="maxLength">
      <number>15</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>20</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none;
font-size:20px;
</string>
     </property>
     <property name="text">
      <string>source IP</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_dPort">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>120</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:white;
border:none;
padding-left:5px;
color:black;</string>
     </property>
     <property name="maximum">
      <number>99999</number>
     </property>
    </widget>
    <zorder>comboBoxChain</zorder>
    <zorder>comboBoxOption</zorder>
    <zorder>comboBoxAction</zorder>
    <zorder>label_2</zorder>
    <zorder>label_3</zorder>
    <zorder>label_4</zorder>
    <zorder>label_5</zorder>
    <zorder>btn_create</zorder>
    <zorder>label_8</zorder>
    <zorder>comboBoxProtocol</zorder>
    <zorder>label_9</zorder>
    <zorder>label_10</zorder>
    <zorder>txt_destIp</zorder>
    <zorder>spinBox_sPort</zorder>
    <zorder>label_11</zorder>
    <zorder>txt_sourceIp</zorder>
    <zorder>spinBox_dPort</zorder>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>420</y>
      <width>741</width>
      <height>391</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border:3px solid white;
font-size:15px;
background-color:#336699;
font:white;</string>
    </property>
    <property name="title">
     <string>READ OR DELETE RULE</string>
    </property>
    <widget class="QComboBox" name="comboBoxRead">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>40</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none;
color:black;
background-color:white;
</string>
     </property>
     <property name="frame">
      <bool>false</bool>
     </property>
     <item>
      <property name="text">
       <string>INPUT</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>OUTPUT</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>FORWARD</string>
      </property>
     </item>
    </widget>
    <widget class="QPushButton" name="btn_display">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>40</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:#ff9900;
border:none;
border-radius:10px;
color:white;</string>
     </property>
     <property name="text">
      <string>DISPLAY</string>
     </property>
    </widget>
    <widget class="QScrollArea" name="scrollArea">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>90</y>
       <width>661</width>
       <height>181</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none;
background-color:#85adad;

</string>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>661</width>
        <height>181</height>
       </rect>
      </property>
      <widget class="QTableWidget" name="tableWidget">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>651</width>
         <height>181</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>651</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::ContiguousSelection</enum>
       </property>
       <attribute name="horizontalHeaderDefaultSectionSize">
        <number>208</number>
       </attribute>
       <attribute name="verticalHeaderDefaultSectionSize">
        <number>38</number>
       </attribute>
       <row>
        <property name="text">
         <string>1</string>
        </property>
       </row>
       <row>
        <property name="text">
         <string>2</string>
        </property>
       </row>
       <column>
        <property name="text">
         <string>Target</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Protocol</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Source</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Destination</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>SourcePort</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>DestPort</string>
        </property>
       </column>
       <item row="0" column="0">
        <property name="text">
         <string/>
        </property>
        <property name="flags">
         <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsDropEnabled|ItemIsUserCheckable</set>
        </property>
       </item>
      </widget>
     </widget>
    </widget>
    <widget class="QPushButton" name="btn_delete">
     <property name="geometry">
      <rect>
       <x>610</x>
       <y>330</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:none;
background-color:#cc0000;
border-radius:10px;
color:white;</string>
     </property>
     <property name="text">
      <string>DELETE</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxSelection">
     <property name="geometry">
      <rect>
       <x>491</x>
       <y>331</y>
       <width>61</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color:white;
border:none;
padding-left:5px;
color:black;</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>250</y>
       <width>661</width>
       <height>61</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font-size:25px;
border:none;
font-weight:bold;</string>
     </property>
     <property name="text">
      <string>_______________________________________________________</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>330</y>
       <width>301</width>
       <height>21</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font-size:13px;
border:none;</string>
     </property>
     <property name="text">
      <string>select the index of the rule you wish to delete</string>
     </property>
    </widget>
    <zorder>comboBoxRead</zorder>
    <zorder>btn_display</zorder>
    <zorder>btn_delete</zorder>
    <zorder>spinBoxSelection</zorder>
    <zorder>label_7</zorder>
    <zorder>scrollArea</zorder>
    <zorder>label_12</zorder>
   </widget>
   <widget class="QPushButton" name="btn_flush">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>100</y>
      <width>91</width>
      <height>33</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: black;
color: white;</string>
    </property>
    <property name="text">
     <string>FLUSH</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_save">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>100</y>
      <width>91</width>
      <height>33</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: black;
color: white;</string>
    </property>
    <property name="text">
     <string>SAVE</string>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
